docs/content/preview/reference/drivers/go/yb-pgx-reference.md

```diff
@@ -0,0 +1,308 @@
+
+```sql
+CREATE TABLE employee (id int PRIMARY KEY, name varchar, age int, language varchar)
+```
+
+```go
+var createStmt = 'CREATE TABLE employee (id int PRIMARY KEY,
+                  name varchar, age int, language varchar)';
+_, err = conn.Exec(context.Background(), createStmt)
+if err != nil {
+  fmt.Fprintf(os.Stderr, "Exec for create table failed: %v\n", err)
+}
+```
+
+The `conn.Exec()` function also returns an `error` object which, if not `nil`, needs to be handled
+in your code.
+
+Read more on designing [Database schemas and tables](../../../../explore/ysql-language-features/databases-schemas-tables/).
+
+### Read and write data
+
+#### Insert data
+
+To write data into YugabyteDB, execute the `INSERT` statement using the same `conn.Exec()`
+function.
+
+```sql
+INSERT INTO employee(id, name, age, language) VALUES (1, 'John', 35, 'Go')
+```
+
+```go
+var insertStmt string = "INSERT INTO employee(id, name, age, language)" +
+                        " VALUES (1, 'John', 35, 'Go')";
+_, err = conn.Exec(context.Background(), insertStmt)
+if err != nil {
+  fmt.Fprintf(os.Stderr, "Exec for create table failed: %v\n", err)
+}
+```
+
+By default, the YugabyteDB PGX driver automatically prepares and caches statements.
+
+#### Query data
+
+To query data from YugabyteDB tables, execute the `SELECT` statement using the function `conn.Query()`.
+Query results are returned in `pgx.Rows` which can be iterated using `pgx.Rows.next()` method.
+Then read the data using `pgx.rows.Scan()`.
+
+The SELECT DML statement:
+
+```sql
+SELECT * from employee;
+```
+
+Code snippet:
+
+```go
+var name string
+var age int
+var language string
+
+rows, err := conn.Query(context.Background(), "SELECT name, age, language FROM employee WHERE id = 1")
+if err != nil {
+  log.Fatal(err)
+}
+defer rows.Close()
+
+fmt.Printf("Query for id=1 returned: ");
+for rows.Next() {
+  err := rows.Scan(&name, &age, &language)
+  if err != nil {
+    log.Fatal(err)
+  }
+  fmt.Printf("Row[%s, %d, %s]\n", name, age, language)
+}
+
+err = rows.Err()
+if err != nil {
+  log.Fatal(err)
+}
+```
+
+### Using pgxpool API
+
+The YugabyteDB PGX driver also provides pool APIs via the `pgxpool` package. You can import it as follows:
+
+```go
+import (
+  "github.com/yugabyte/pgx/tree/master/pgxpool"
+)
+```
+
+#### Establishing a connection
+
+The primary way of establishing a connection is with `pgxpool.Connect()`.
+
+```go
+pool, err := pgxpool.Connect(context.Background(), os.Getenv("DATABASE_URL"))
+```
+
+You can also provide configuration for the pool as follows:
+
+```go
+config, err := pgxpool.ParseConfig(os.Getenv("DATABASE_URL"))
+if err != nil {
+    // ...
+}
+config.AfterConnect = func(ctx context.Context, conn *pgx.Conn) error {
+    // do something with every new connection
+}
+
+pool, err := pgxpool.ConnectConfig(context.Background(), config)
+```
+
```

Comment:
We can add below content here:

You can either `Acquire` a connection from pool and execute queries on it or use `Query` API to directly execute SQLs on the pool.

```go
conn, err := pool.Acquire(context.Background())
if err != nil {
	fmt.Fprintf(os.Stderr, "Unable to connect to database: %v\n", err)
	os.Exit(1)
}
defer conn.Release()

var createStmt = `CREATE TABLE employee (id int PRIMARY KEY,
                  name varchar, age int, language varchar)`
_, err = conn.Exec(context.Background(), createStmt)

// ...

rows, err := pool.Query(context.Background(), "SELECT name, age, language FROM employee WHERE id = 1")
```

Reply:
Thanks . Will include and modify the rest of the steps as per for feedback.